performance/query_processing.rst

```diff
@@ -53,6 +63,179 @@ The task tracker executor is well suited for long running, complex data warehous
 
 Each task tracker daemon on the workers also makes sure to execute at most citus.max_running_tasks_per_node concurrently. This concurrency limit helps in avoiding disk I/O contention when queries are not served from memory. The task tracker executor is designed to efficiently handle complex queries which require repartitioning and shuffling intermediate data among workers.
 
+.. _push_pull_execution:
+
+Subquery/CTE Push-Pull Execution
+--------------------------------
+
+If necessary Citus can gather results from subqueries and CTEs into the coordinator node and then push them back across workers for use by an outer query. This allows Citus to support a greater variety of SQL constructs, and even mix executor types between a query and its subqueries.
+
+For example, having subqueries in a WHERE clause sometimes cannot execute inline at the same time as the main query, but must be done separately. Suppose a web analytics application maintains a ``visits`` table partitioned by ``page_id``. To query the number of visitor sessions on the top twenty most visited pages, we can use a subquery to find the list of pages, then an outer query to count the sessions.
+
+.. code-block:: sql
+
+  SELECT page_id, count(distinct session_id)
+  FROM visits
+  WHERE page_id IN (
+    SELECT page_id
+    FROM visits
+    GROUP BY page_id
+    ORDER BY count(*) DESC
+    LIMIT 20
+  )
+  GROUP BY page_id;
+
+The real-time executor would like to run a fragment of this query against each shard by page_id, counting distinct session_ids, and combining the results on the coordinator. However the LIMIT in the subquery means the subquery cannot be executed as part of the fragment. By recursively planning the query Citus can run the subquery separately, push the results to all workers, run the main fragment query, and pull the results back to the coordinator. The "push-pull" design supports a subqueries like the one above.
+
+Let's see this in action by reviewing the EXPLAIN output. First we'll try the query and remove the LIMIT clause. Then we'll add the LIMIT clause back and see what happens to the execution plan.
```

Comment:
So if I understand this right, the first example should get pushed down entirely to the workers? The second example with the LIMIT added back will use the pull-push planner? If so, I'd skip the first example, and go directly to the one with LIMIT. I do see value in comparing them, but we haven't made it explicit or clear that this query gets pushed down.